controllers/spifilecontentrequest_controller.go

```diff
@@ -154,13 +154,24 @@ func (r *SPIFileContentRequestReconciler) Reconcile(ctx context.Context, req ctr
 	}
 
 	if request.Status.Phase == "" {
+		// check if the URL is processable, otherwise fail fast
+		sp, _ := r.ServiceProviderFactory.FromRepoUrl(ctx, request.Spec.RepoUrl)
+		_, ok := sp.(serviceprovider.ScmProvider)
+		if !ok {
+			r.updateFileRequestStatusError(ctx, &request, serviceprovider.FileDownloadNotSupportedError{})
+			return ctrl.Result{}, serviceprovider.FileDownloadNotSupportedError{}
+		}
 		request.Status.Phase = api.SPIFileContentRequestPhaseAwaitingTokenData
 	} else if request.Status.Phase == api.SPIFileContentRequestPhaseDelivered {
 		// already injected, nothing to do
 		return ctrl.Result{}, nil
 	}
 
 	if request.Status.LinkedBindingName == "" {
+		if request.Status.Phase == api.SPIFileContentRequestPhaseError {
```

Comment:
Is it possible that we can recover from the state where we set the phase to error? if so, we should not short-circuit here...